File: runtime/src/main/kotlin/de/fatox/meta/ui/components/MetaValidTextField.java
package de.fatox.meta.ui.components;

import com.kotcrab.vis.ui.widget.VisLabel;
import com.kotcrab.vis.ui.widget.VisValidatableTextField;

public class MetaValidTextField {
    private VisLabel descriptionLabel;
    private VisLabel errorLabel;

    private VisValidatableTextField textField;

    public MetaValidTextField(String description, VisLabel errorLabel) {
        descriptionLabel = new VisLabel(description);
        textField = new VisValidatableTextField();
        this.errorLabel = errorLabel;
    }

    public VisValidatableTextField getTextField() {
        return textField;
    }

    public VisLabel getDescription() {
        return descriptionLabel;
    }

    public void addValidator(MetaInputValidator inputValidator) {
        inputValidator.setErrorLabel(errorLabel);
        textField.addValidator(inputValidator);
    }
}

File: runtime/src/main/kotlin/de/fatox/meta/ui/components/MetaValidTextField.kt
package de.fatox.meta.ui.components

import de.fatox.meta.ui.MetaType

class MetaValidTextField(description: String, private val errorLabel: MetaLabel) {
	val description: MetaLabel = MetaLabel(description, MetaType.CAPTION)

	val textField: MetaValidatableTextField = MetaValidatableTextField()

	fun addValidator(inputValidator: MetaInputValidator) {
		inputValidator.errorLabel = errorLabel
		textField.addValidator(inputValidator)
	}
}
